PWGHF/TableProducer/treeCreatorD0ToKPi.cxx

```diff
@@ -460,7 +463,61 @@ struct HfTreeCreatorD0ToKPi {
+                                     MatchedGenCandidatesMc const& particles,
+                                     TracksWPid const& tracks, aod::BCs const& BCs)
+  {
+    processMc<0, true>(collisions, McCollisions, candidates, reconstructedCandBkg, particles, tracks, BCs);
+  }
+  void processMcWithDCAFitterAll(aod::Collisions const& collisions,
+                                 aod::McCollisions const& McCollisions,
+                                 SelectedCandidatesMc const& candidates,
+                                 MatchedGenCandidatesMc const& particles,
+                                 TracksWPid const& tracks, aod::BCs const& BCs)
+  {
+    processMc<0>(collisions, McCollisions, candidates, candidates, particles, tracks, BCs);
+  }
+  void processMcWithKFParticleOnlySig(aod::Collisions const& collisions,
+                                      aod::McCollisions const& McCollisions,
+                                      SelectedCandidatesMcKF const& candidates,
+                                      MatchedGenCandidatesMc const& particles,
+                                      TracksWPid const& tracks, aod::BCs const& BCs)
+  {
+    processMc<1>(collisions, McCollisions, candidates, reconstructedCandSigKF, particles, tracks, BCs);
+  }
+  void processMcWithKFParticleOnlyBkg(aod::Collisions const& collisions,
+                                      aod::McCollisions const& McCollisions,
+                                      SelectedCandidatesMcKF const& candidates,
+                                      MatchedGenCandidatesMc const& particles,
+                                      TracksWPid const& tracks, aod::BCs const& BCs)
+  {
+    processMc<1, true>(collisions, McCollisions, candidates, reconstructedCandBkgKF, particles, tracks, BCs);
+  }
+  void processMcWithKFParticleAll(aod::Collisions const& collisions,
+                                  aod::McCollisions const& McCollisions,
+                                  SelectedCandidatesMcKF const& candidates,
+                                  MatchedGenCandidatesMc const& particles,
+                                  TracksWPid const& tracks, aod::BCs const& BCs)
+  {
+    processMc<1>(collisions, McCollisions, candidates, candidates, particles, tracks, BCs);
+  }
+
+  PROCESS_SWITCH(HfTreeCreatorD0ToKPi, processMcWithDCAFitterOnlySig, "Process MC with DCAFitterN only for signals", false);
+  PROCESS_SWITCH(HfTreeCreatorD0ToKPi, processMcWithDCAFitterOnlyBkg, "Process MC with DCAFitterN only for background", false);
+  PROCESS_SWITCH(HfTreeCreatorD0ToKPi, processMcWithDCAFitterAll, "Process MC with DCAFitterN", false);
+  PROCESS_SWITCH(HfTreeCreatorD0ToKPi, processMcWithKFParticleOnlySig, "Process MC with KFParticle only for signals", false);
+  PROCESS_SWITCH(HfTreeCreatorD0ToKPi, processMcWithKFParticleOnlyBkg, "Process MC with KFParticle only for background", false);
+  PROCESS_SWITCH(HfTreeCreatorD0ToKPi, processMcWithKFParticleAll, "Process MC with KFParticle", false);
```

Comment:
Thanks! I added a check to make sure only one process is enabled at a time and removed the obsolete configurable.
But I kept the default settings of these processes to be `false` because these are MC processes. the `processDataWithDCAFitterN` is set to be the default `true`